### PR TITLE
Jammy support

### PR DESCRIPTION
**Pull request by addyess**
* adding jammy support for this charm
* adding extra black coverage
* updates to the snapshot action for jammy dealing with snapd confinement
* updates testing to allow for adjusting the series of the etcd charm during deployment:

    ```bash
    tox -e integration -- --series=jammy
    ```

### CODE REVIEW

File: actions/snapshot [hunk 1]
```diff
@@ -24,11 +24,13 @@ DATE_STAMP=$(date +%Y-%m-%d-%H.%M.%S)
...
 if [ "${ETCD_KEYS_VERSION}" == "v2" ]; then
-  # Dump the data currently in the cluster
-  ETCDCTL_API=2 /snap/bin/etcd.etcdctl backup --data-dir $ETCD_DATA_DIR --backup-dir $ETCD_BACKUP_TARGET_DIR/$JUJU_ACTION_UUID
+  # Dump the data currently in the cluster from the snap, then move to $ETCD_BACKUP_TARGET_DIR
+  V2_SNAPSHOT=$ETCD_DATA_DIR/v2-snapshots/$JUJU_ACTION_UUID
+  ETCDCTL_API=2 /snap/bin/etcd.etcdctl backup --data-dir $ETCD_DATA_DIR --backup-dir $V2_SNAPSHOT
+  mv $V2_SNAPSHOT $ETCD_BACKUP_TARGET_DIR/
 elif [ "${ETCD_KEYS_VERSION}" == "v3" ]; then
   mkdir -p $ETCD_BACKUP_TARGET_DIR/$JUJU_ACTION_UUID
   cp $ETCD_DATA_DIR/member/snap/db $ETCD_BACKUP_TARGET_DIR/$JUJU_ACTION_UUID
```
Comment (addyess) [inline actions/snapshot:36]: careful `bash` adjustment here to make up for the fact that in jammy, snapd confinement is more serious about not letting `etcdctl` write to just anywhere.  Now for v2 snapshots, this script will write to the `/var/snap/etcd/v2-snapshots/` directory, then move that backup to the target directory after the fact.

File: actions/restore.py [hunk 2]
```diff
@@ -1,9 +1,7 @@
 #!/usr/local/sbin/charm-env python3
 
-import random
 from charms import layer
-from charms.templating.jinja2 import render
```
Comment (addyess) [inline actions/restore.py:5]: switching to `charmhelpers.core.templating`

File: reactive/etcd.py [hunk 3]
```diff
@@ -18,7 +18,7 @@
 from charms.reactive import register_trigger
 from charms.reactive.helpers import data_changed
 
-from charms.templating.jinja2 import render
+from charmhelpers.core.templating import render
```
Comment (addyess) [inline reactive/etcd.py:21]: switching to `charmhelpers.core.templating`

**Comment by hyperbolic2346** [comment]
Hard to see the change for all the ' -> " work here. Are single quotes out of fashion now?

**Comment by addyess** [comment]
> Hard to see the change for all the ' -> " work here. Are single quotes out of fashion now?

yes, the addition of `black` in the `actions/` path causes the majority of these changes

**Review by berkayoz — APPROVED** [review]
Great work, LGTM. I don't know a lot about `charmhelpers` so I assume there is no difference between `charmhelpers.core.templating` and `charms.templating.jinja2`.

**Review by Cynerva — APPROVED** [review]

**Pull request merged**